pyat/at/lattice/transformation.py

```diff
@@ -0,0 +1,280 @@
+        ])
+    T0 = numpy.array([
+        ld * r3d[2, 0] / r3d[2, 2], r3d[2, 0], ld * r3d[2, 1] / r3d[2, 2],
+        r3d[2, 1], 0, ld / r3d[2, 2]
+    ])
+    return T0 + tD0
+
+def _r_matrix(ld, r3d):
+    """
+    The implementation follows the one described in:
+    https://doi.org/10.1016/j.nima.2022.167487
+    All the comments featuring 'Eq' points to the paper's equations.
+    
+    Rotation matrix operator (R1, R2).
+    Can take into account the effect of a longitudinal displacement.
+    ld: Longitudinal displacement [m]
+    r3d: 3D rotation matrix
+    Corresponds to Eq. (9)
+    """
+    return numpy.array([
+        [
+            r3d[1, 1] / r3d[2, 2],
+            ld * r3d[1, 1] / r3d[2, 2] ** 2,
+            -r3d[0, 1] / r3d[2, 2],
+            -ld * r3d[0, 1] / r3d[2, 2] ** 2,
+            0,
+            0,
+        ],
+        [
+            0,
+            r3d[0, 0],
+            0,
+            r3d[1, 0],
+            r3d[2, 0],
+            0,
+        ],
+        [
+            -r3d[1, 0] / r3d[2, 2],
+            -ld * r3d[1, 0] / r3d[2, 2] ** 2,
+            r3d[0, 0] / r3d[2, 2],
+            ld * r3d[0, 0] / r3d[2, 2] ** 2,
+            0,
+            0,
+        ],
+        [
+            0,
+            r3d[0, 1],
+            0,
+            r3d[1, 1],
+            r3d[2, 1],
+            0,
+        ],
+        [
+            0,
+            0,
+            0,
+            0,
+            1,
+            0,
+        ],
+        [
+            -r3d[0, 2] / r3d[2, 2],
+            -ld * r3d[0, 2] / r3d[2, 2] ** 2,
+            -r3d[1, 2] / r3d[2, 2],
+            -ld * r3d[1, 2] / r3d[2, 2] ** 2,
+            0,
+            1,
+        ],
+    ])
+    
+def transform_elem(elem: Element, midpoint: str = "center",
+                   dx: float = 0.0, dy: float = 0.0,                       
+                   tilt: float = 0.0, pitch: float = 0.0, yaw: float = 0.0,
+                   relative: bool = False) -> None:
+    r"""Set the tilt, pitch and yaw angle of an :py:class:`.Element`.
+    The tilt is a rotation around the *s*-axis, the pitch is a
+    rotation around the *x*-axis and the yaw is a rotation around
+    the *y*-axis.
+
+    A positive angle represents a clockwise rotation when
+    looking in the direction of the rotation axis.
+
+    The transformations are not all commmutative. The rotations are applied in 
+    the order *Z* -> *Y* -> *X* (tilt -> yaw -> pitch). The element is rotated 
+    around its mid-point. The mid-point can either be the element entrance 
+    (entry face of the downstream drift element) or center (axis joining the 
+    entry and exit points of the element).
+
+    If *relative* is :py:obj:`True`, the previous angles are rebuilt from the 
+    *r3d* matrix and incremented by the input arguments.
+    *relative* only allows to add the previous angles, not the transverse 
+    shifts.
+    The shift is always conserved regardless of the value of *relative*.
```

Comment:
> The shift is always conserved regardless of the value of *relative*.

This sentence is not clear. Could it be "The shift is always absolute regardless of the value of *relative*"?